_drafts/release-workflows.md

```diff
@@ -0,0 +1,309 @@
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Default branch      | Git branch that contains the stable code (made available to all users). Usually named `master`.                                                                                    |
+| Future branches     | Permanent Git branch that contains the unstable code (made available only to early adopters). Usually named `next`.                                                                |
+| Channel             | Distribution channels used to limit the accessibility of a release to a subset of users. For example `dist-tags` in `npm`.                                                         |
+| Default channel     | Distribution channel to make stable version available to every users. Usually named `latest`.                                                                                      |
+| Future channels     | Distribution channels to make unstable version available to early adopters. Usually named `next`.                                                                                  |
+| Pre-release version | version that doesn't follow the regular `semver` increment pattern. For example any increment of `2.0.0-beta.1` is `2.0.0-beta.2`, then `2.0.0-beta.2` etc...                      |
+| Version             | version following the `semver` rules. For example, a `minor` increment of `1.0.0` is `1.1.0`.                                                                                      |
+| Module              | A software meant to be used as a dependency of other software. It's not meant to be used by final users.                                                                           |
+| Application         | A software meant to be used by final users. No other software depends on it.                                                                                                       |                                                         |
+
+### Branch types
+
+#### Single release branch
+
+Simple workflow publishing release for each commits on the `master` branch.
+
+Ideal for small or stable modules receiving few breaking changes.
+
+Branch: `master`
+
+Channel: `latest` if supported by the release target
+
+##### Single release branch workflows
+
+| Action                               | Result                                                          |
+|--------------------------------------|-----------------------------------------------------------------|
+| **push** to `master`                 | Release on default channel (if supported, otherwise no channel) |
+| **merge** feature branch to `master` | Release on default channel (if supported, otherwise no channel) |
+
+##### Single release branch configuration
+
+```json
+{
+  "release": {
+    "branches": ["master"]
```

Comment:
Can this be inferred from the project's SCM platform?

Reply:
I'm not sure such feature is worthwhile or even a good idea for a few reasons:
- It won't be possible to support all SCM. GitHub, GitLAb and probably Bitbucket have an API for that but not other SCM, in particular self hosted ones
- It would create a dependency between semantic-release core and specific SCM API. Currently all SCM specific stuff is handle in respective plugins and I'd like to keep this separation
- It would be additional code to write, test and maintain that would just prevent a small percentage of user to configure the branch name. Most users would use `master` as default. And a lot of user would have to overwrite the `branches` property anyway to implement their workflow